langlib/lang.regexp/src/main/ballerina/regexp.bal

```diff
@@ -22,13 +22,27 @@ import ballerina/jballerina.java;
 @builtinSubtype
 public type RegExp any;
 
+# A span of a string.
+# A span is a substring of another string.
 public type Span readonly & object {
+    # The index within the string where the span starts.
     public int startIndex;
+    # The index within the string following the end of the span.
+    # The length of the span is `endIndex - startIndex`.
     public int endIndex;
-    // This avoids constructing a potentially long string unless and until it is needed.
+    # Returns a string with the content of the span.
     public isolated function substring() returns string;
 };
 
+# A list providing detailed information about the match of a regular expression within string.
+# Each member of the list identifies the `Span` within the string matched
+# by each of the regular expression's capturing groups.
+# The member with index 0 corresponds to the entire regular expression.
+# The group with index i, where i > 1,is the i-th capturing group;
+# this will be nil if the match of the regular expression did not use
+# a match of the capturing group.
+# The capturing groups within a regular expression are ordered by the position
+# of their opening parenthesis.
 public type Groups readonly & [Span, Span?...];
 
 type SpanAsTupleType [int, int, string];
```

Comment:
Shall we add small description for this type as well?

Reply:
Those types are internal types we maintain to map between the public types. IMO, the name is self-explanatory